platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```
@@ -382,12 +382,30 @@ export default class HangingProtocolService extends PubSubService {
       return;
     }
 
-    const matchedProtocol = this.protocolEngine.run({
+    let matchedProtocol = this.protocolEngine.run({
       studies: this.studies,
       activeStudy,
       displaySets,
     });
-    this._setProtocol(matchedProtocol);
+    let nthBestMatch = 1;
```

You should just throw the exception if a specific protocol is specified